src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1640,15 +1640,30 @@ void validateParamsType(BindingDiagnosticBag diagnostics)
                                 return;
                             }
 
-                            MethodSymbol? collectionBuilderMethod = binder.GetAndValidateCollectionBuilderMethod(syntax, (NamedTypeSymbol)Type, diagnostics, elementType: out _);
-                            if (collectionBuilderMethod is null)
-                            {
-                                return;
-                            }
+                            var targetType = (NamedTypeSymbol)Type;
+                            targetType.OriginalDefinition.HasCollectionBuilderAttribute(out TypeSymbol? builderType, out string? methodName);
+                            Debug.Assert(builderType is { });
+                            Debug.Assert(!string.IsNullOrEmpty(methodName));
 
-                            if (ContainingSymbol.ContainingSymbol is NamedTypeSymbol) // No need to check for lambdas or local function
+                            var candidateMethods = binder.GetAndValidateCollectionBuilderMethods(syntax, targetType.OriginalDefinition, builderType, methodName, diagnostics);
+                            if (candidateMethods.Any())
                             {
-                                checkIsAtLeastAsVisible(syntax, binder, collectionBuilderMethod, diagnostics);
+                                var useSiteInfo = new CompoundUseSiteInfo<AssemblySymbol>(diagnostics, ContainingAssembly);
+                                var spanType = DeclaringCompilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T).Construct(elementType);
+                                var typeArguments = targetType.GetAllTypeArguments(ref useSiteInfo);
+                                diagnostics.Add(syntax, useSiteInfo);
+                                var candidateMethodGroup = binder.BindCollectionBuilderMethodGroup(syntax, methodName, typeArguments, candidateMethods);
+                                var collectionCreation = binder.BindCollectionBuilderCreate(
```

Comment:
What happens if we have a single Create method which takes a ROS and an optional parameter? The params-collections spec says this would not be an applicable Create method and I think the code did that (it had a check that we only have one parameter), but the new logic was relaxed to allow multiple parameters.
Do we need to update the spec with regards to `params`?